OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/FileActions.java

```diff
@@ -0,0 +1,80 @@
+/* 
+ * Copyright (c) 2015 Sebastian Brudzinski
+ * 
+ * See the file LICENSE for copying permission.
+ */
+package latexstudio.editor;
+
+import java.io.File;
+import java.io.InputStream;
+import latexstudio.editor.files.FileChooserService;
+import latexstudio.editor.files.FileService;
+
+/**
+ * This class provides common editor actions
+ *
+ * @author maciej
+ */
+public class FileActions {
+
+    private EditorTopComponent etc;
+    private static final ApplicationLogger LOGGER = new ApplicationLogger("File Actions");
+
+    public FileActions(EditorTopComponent etc) {
+        this.etc = etc;
+    }
+
+    /**
+     * Opens file content in EditorTopComponent
+     *
+     * @param file file, that exists and isn't a folder
+     */
+    public void openFile(File file) {
```

Comment:
Question: Is this a background task?

This could last very long if reading from IO takes very long.
Maybe this should be a new todo in class FileService?